src/openrct2/windows/view_clipping.c

```diff
@@ -0,0 +1,286 @@
+		break;
+	case WIDX_CLIP_HEIGHT_CHECKBOX:
+		// Toggle height clipping.
+		mainWindow = window_get_main();
+		if (mainWindow != NULL) {
+			mainWindow->viewport->flags ^= VIEWPORT_FLAG_PAINT_CLIP_TO_HEIGHT;
+			window_invalidate(mainWindow);
+		}
+		window_invalidate(w);
+		break;
+	case WIDX_CLIP_HEIGHT_INCREASE:
+		if (gClipHeight < 255)
+			window_view_clipping_set_clipheight(w, gClipHeight + 1);
+		mainWindow = window_get_main();
+		if (mainWindow != NULL)
+			window_invalidate(mainWindow);
+		break;
+	case WIDX_CLIP_HEIGHT_DECREASE:
+		if (gClipHeight > 0)
+			window_view_clipping_set_clipheight(w, gClipHeight - 1);
+		mainWindow = window_get_main();
+		if (mainWindow != NULL)
+			window_invalidate(mainWindow);
+		break;
+	case WIDX_CLIP_HEIGHT_VALUE:
+		// Toggle display of the cut height value in RAW vs UNITS
+		gClipHeightDisplayType = !gClipHeightDisplayType;
+		window_invalidate(w);
+		break;
+	}
+}
+
+static void window_view_clipping_update(rct_window *w)
+{
+	const rct_widget *const widget = &window_view_clipping_widgets[WIDX_CLIP_HEIGHT_SLIDER];
+	const rct_scroll *const scroll = &w->scrolls[0];
+	const sint16 scroll_width = widget->right - widget->left - 1;
+	const uint8 clip_height = (uint8)(((float)scroll->h_left / (scroll->h_right - scroll_width)) * 255);
+	if (clip_height != gClipHeight) {
+		gClipHeight = clip_height;
+
+		// Update the main window accordingly.
+		rct_window *mainWindow = window_get_main();
+		if (mainWindow != NULL) {
+			window_invalidate(mainWindow);
+		}
+	}
+	widget_invalidate(w, WIDX_CLIP_HEIGHT_SLIDER);
+}
+
+static void window_view_clipping_invalidate(rct_window *w)
+{
+	colour_scheme_update(w);
+
+	widget_scroll_update_thumbs(w, WIDX_CLIP_HEIGHT_SLIDER);
+
+	rct_window *mainWindow = window_get_main();
+	if (mainWindow != NULL) {
+		widget_set_checkbox_value(w, WIDX_CLIP_HEIGHT_CHECKBOX, mainWindow->viewport->flags & VIEWPORT_FLAG_PAINT_CLIP_TO_HEIGHT);
+	}
+}
+
+static void window_view_clipping_paint(rct_window *w, rct_drawpixelinfo *dpi)
+{
+	window_draw_widgets(w, dpi);
+
+	// Clip height value
+	sint32 x = w->x + 8;
+	sint32 y = w->y + w->widgets[WIDX_CLIP_HEIGHT_VALUE].top;
+	gfx_draw_string_left(dpi, STR_VIEW_CLIPPING_HEIGHT_VALUE, NULL, w->colours[0], x, y);
+
+	x = w->x + w->widgets[WIDX_CLIP_HEIGHT_VALUE].left + 1;
+	y = w->y + w->widgets[WIDX_CLIP_HEIGHT_VALUE].top;
+
+	fixed32_2dp clipHeightValueInMeters;
+	fixed16_1dp clipHeightValueInFeet;
+	sint32 clipHeightRawValue = (sint32)gClipHeight;
+	switch (gClipHeightDisplayType) {
+	case DISPLAY_RAW:
+	default:
+		gfx_draw_string_left(dpi, STR_FORMAT_INTEGER, &clipHeightRawValue, w->colours[0], x, y); //Printing the raw value.
+		break;
+
+	case DISPLAY_UNITS:
+		// Print the value in the configured measurement units.
+		switch (gConfigGeneral.measurement_format) {
+		case MEASUREMENT_FORMAT_METRIC:
+		case MEASUREMENT_FORMAT_SI:
+			clipHeightValueInMeters = (fixed32_2dp)(FIXED_2DP(gClipHeight, 0) / 2 * 1.5 - FIXED_2DP(10, 50));
```

Comment:
Can you suffix the real values with `f`? otherwise they will needlessly be promoted to `double`s